src/node_contextify.cc

```diff
@@ -702,13 +749,69 @@ void ContextifyContext::PropertyEnumeratorCallback(
   if (IsStillInitializing(ctx)) return;
 
   Local<Array> properties;
-
-  if (!ctx->sandbox()->GetPropertyNames(ctx->context()).ToLocal(&properties))
+  // Only get named properties, exclude symbols and indices.
+  if (!ctx->sandbox()
+           ->GetPropertyNames(
+               ctx->context(),
+               KeyCollectionMode::kIncludePrototypes,
+               static_cast<PropertyFilter>(PropertyFilter::ONLY_ENUMERABLE |
+                                           PropertyFilter::SKIP_SYMBOLS),
+               IndexFilter::kSkipIndices)
+           .ToLocal(&properties))
     return;
 
   args.GetReturnValue().Set(properties);
 }
 
+// static
+void ContextifyContext::IndexedPropertyEnumeratorCallback(
+    const PropertyCallbackInfo<Array>& args) {
+  Isolate* isolate = args.GetIsolate();
+  HandleScope scope(isolate);
+  ContextifyContext* ctx = ContextifyContext::Get(args);
+  Local<Context> context = ctx->context();
+
+  // Still initializing
+  if (IsStillInitializing(ctx)) return;
+
+  Local<Array> properties;
+
+  if (!ctx->sandbox()->GetPropertyNames(context).ToLocal(&properties)) return;
```

Comment:
Shouldn't we just use the variant of GetPropertyNames with a property filter that skips strings and symbols + KeyConversionMode::kKeepNumbers?

Reply:
`PropertyFilter::SKIP_STRINGS` would filter both strings including indices. It is the counter argument to `PropertyFilter::SKIP_SYMBOLS`.

Reply:
Oh I see, maybe just PropertyFilter::SKIP_SYMBOLS and KeyConversionMode::kKeepNumbers?

Reply:
Removed unnecessary type conversion in https://github.com/nodejs/node/pull/53517/commits/dd8c0c64c3aab01c6758e7ac676f3cdda7b6bbdf.